src/quest_manager/views.py

```diff
@@ -184,17 +184,25 @@ class QuestCopy(QuestCreate):
     def get_form_kwargs(self):
         kwargs = super().get_form_kwargs()
 
+        quest = get_object_or_404(Quest, pk=self.kwargs['quest_id'])
+
+        kwargs['initial'] = model_to_dict(quest)
+        kwargs['initial']['tags'] = quest.tags.all()
+
         # by default, set the quest this was copied from as the new_quest_prerequisite
         # If this is changed in the form it will be overwritten in form_valid() from QuestFormViewMixin
-        copied_quest = get_object_or_404(Quest, pk=self.kwargs['quest_id'])
-        kwargs['initial']['new_quest_prerequisite'] = copied_quest
+        kwargs['initial']['new_quest_prerequisite'] = quest
+
+        # rename copied quest
+        # strips front if len(name + ' - copy') is too long
+        char_limit = self.model._meta.get_field('name').max_length
+        copy_text = ' - Copy'
+        name = quest.name
 
-        new_quest = get_object_or_404(Quest, pk=self.kwargs['quest_id'])
-        new_quest.pk = None  # autogen a new primary key (quest_id by default)
-        new_quest.import_id = uuid.uuid4()
```

Comment:
Please revert to old copy code unless there's a specific reason for changing.  Also, please add test to check that a copied quest has a unique UUID!!  Does this pass tests?  I thought that field was unique so I'm surprised the form can even be submitted without a new UUID, unless it's auto generated in the background?

Reply:
no need to create a new instance of quest (old code is also calling get_object_or_404(Quest, pk=self.kwargs['quest_id']) twice)

i think i just forgot to get new uuid. since i deleted everything in old code because i had a dynamic function that would make a new instance of a model in ``TagModelMixin``